src/WCSimPrimaryGeneratorAction.cc

```diff
@@ -1552,6 +1558,63 @@ void WCSimPrimaryGeneratorAction::GeneratePrimaries(G4Event* anEvent)
       SetBeamDir(dir);
       SetBeamPDG(pdg);
 
+    } else if (useHepMC3Evt) {
+
+      G4cout << "Using HepMC3 event" << G4endl;
+      // Check if the WCSimNuHepMC3Reader object has been initiaited yet
+      if (!hepmc3_reader) {
+        hepmc3_reader = new WCSimNuHepMC3Reader(hepmc3_filename, myDetector);
+      }
+
+      // Get the next event from the reader
+      hepmc3_reader->ReadEvent(hepmc3_positionGen);
+
+      // Loop over the particles
+      for (HepMC3::GenParticlePtr part : hepmc3_reader->event.particles()) {
+
+        // Skip nuclear remnants
+        if (part->pid() == 2009900000) {
+          continue;
+        }
+
+        // If particle has status 4 then it is a beam particle. This needs writing out, but not simulating
+        if (part->status() == 4) {
```

Comment:
you check status
4: beam
1: outgoing

Is there a status x: target? If so, please use that to set
```
targetpdgs[iVertex]
targetenergies[iVertex]
targetdirs[iVertex]
```

Reply:
Added this using `if (part->status() == 20)`